fab/fab/operations/supervisor.py

```diff
@@ -107,7 +107,8 @@ def set_celery_supervisorconf():
             conf_destination_filename = 'supervisor_celery_worker_{}_{}.conf'.format(
                 comma_separated_queue_names, worker_num)
 
-            worker_name = get_celery_worker_name(params['comma_separated_queue_names'],
+            worker_name = get_celery_worker_name(env.env_name,
```

Comment:
@mkangia I just noticed this: `env.env_name` isn't the right variable here, even though most of the time it'll have the same value. What you need is `environment.meta_config.deploy_env`, which I believe in fabric is already set on the fabric `env` variable as `env.deploy_env`.

Reply:
@dannyroberts thanks for flagging.. i think this is correct.
taking example of `icds-new` here, we would want to pass `icds-new` here and not `icds`. The methods i have added, for example `get_celery_worker_name` will call `get_environment` using this env_name passed and hence this looks right to me. Let me know if that does not look right.

i am still confused about now the naming is working.. can you share your experience with that in the README or add some documentation so that one does not trip with this..

Reply:
I'm fairly sure that in the case of `icds-new`, we want the worker name to have `icds` (not `icds-new`) in order to preserve current behavior. Additionally, env_name here is _whatever the user typed into the command line_, so for `icds` where you can use `commcare-cloud icds ...` or `commcare-cloud icds-new ...`, you don't want the outcome to depend on which of those two you typed.

Here is where the `'environment'` param that is used in the current celery template gets set:
https://github.com/dimagi/commcare-cloud/blob/b23dfc47b9790a6ce61428b314dd2c21ac22c95a/fab/fab/operations/supervisor.py#L306
As you can see, it's set to the `deploy_env` value.

I'm not exactly sure where to document, but in general `deploy_env` is used for everything except datadog and email subject lines, which use `env_monitoring_id` (which is usually the same thing, but different for `icds-new`). Both of these are defined in `meta.yml`. In general, we _never_ use `env.env_name` (fab) / `args.environment` (ansible) for anything other than locating the environment configuration, for the reason given above.

Reply:
And thanks for asking, in the course of describing that, I realized there's at least one change I could make to reduce one bit of confusion, and that's to change `args.environment` to `args.env_name` in commcare-cloud so that fab and commcare-cloud call the same value the same thing.

Reply:
Aha. Offlined with @mkangia. The part I was missing is that `get_celery_worker_name` and related functions only use `env.env_name` to look up the environment (which is exactly how it's supposed to be used); it's not used directly in the service name.

Only comment now is that in the meantime `environment.translated_app_processes_config.environment` was moved to `environment.meta_config.deploy_env`, so @mkangia will make that change.

Reply:
thanks @dannyroberts .. 

i think the confusion how things are at fab and commcare-cloud.. is it
1. possible to have it at a single place?
2. follow the same naming convention at both places?

also again taking example of icds-new
icds-new would be the name of the environment
whereas icds would the deploy environment?

i still find that a little confusing but may be thats just a matter of being familiar.. can we use inventory name/inventory instead of environment name? when we talk environment i think one(at least I) always think of this as icds and icds-new.